executor/api/grpc/kfserving/server.go

```diff
@@ -0,0 +1,128 @@
+package kfserving
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/seldonio/seldon-core/executor/api/client"
+	"github.com/seldonio/seldon-core/executor/api/grpc"
+	"github.com/seldonio/seldon-core/executor/api/grpc/kfserving/inference"
+	"github.com/seldonio/seldon-core/executor/api/payload"
+	"github.com/seldonio/seldon-core/executor/predictor"
+	v1 "github.com/seldonio/seldon-core/operator/apis/machinelearning.seldon.io/v1"
+	"net/url"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type GrpcKFServingServer struct {
+	Client    client.SeldonApiClient
+	predictor *v1.PredictorSpec
+	Log       logr.Logger
+	ServerUrl *url.URL
+	Namespace string
+}
+
+func NewGrpcKFServingServer(predictor *v1.PredictorSpec, client client.SeldonApiClient, serverUrl *url.URL, namespace string) *GrpcKFServingServer {
+	return &GrpcKFServingServer{
+		Client:    client,
+		predictor: predictor,
+		Log:       logf.Log.WithName("KFServingGrpcApi"),
+		ServerUrl: serverUrl,
+		Namespace: namespace,
+	}
+}
+
+func (g GrpcKFServingServer) ServerLive(ctx context.Context, request *inference.ServerLiveRequest) (*inference.ServerLiveResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) ServerReady(ctx context.Context, request *inference.ServerReadyRequest) (*inference.ServerReadyResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) ModelReady(ctx context.Context, request *inference.ModelReadyRequest) (*inference.ModelReadyResponse, error) {
+	md := grpc.CollectMetadata(ctx)
+	ctx = context.WithValue(ctx, payload.SeldonPUIDHeader, md.Get(payload.SeldonPUIDHeader)[0])
+	seldonPredictorProcess := predictor.NewPredictorProcess(ctx, g.Client, logf.Log.WithName("infer"), g.ServerUrl, g.Namespace, md)
+	reqPayload := payload.ProtoPayload{Msg: request}
+	resPayload, err := seldonPredictorProcess.Status(&g.predictor.Graph, request.Name, &reqPayload)
+	if err != nil {
+		return nil, err
+	}
+	return resPayload.GetPayload().(*inference.ModelReadyResponse), nil
+}
+
+func (g GrpcKFServingServer) ServerMetadata(ctx context.Context, request *inference.ServerMetadataRequest) (*inference.ServerMetadataResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) ModelMetadata(ctx context.Context, request *inference.ModelMetadataRequest) (*inference.ModelMetadataResponse, error) {
```

Comment:
It seems in the client, model metadata is not implemented, but metadata is - would this not be using th emodelMetadata function instead of the metadata function?

Reply:
As I understand it client modelMetaData is only used by the Seldon specific Graph Metadata call. But I agree this terminology is confusing.